crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+# get the latest checkpoint.
+input CheckpointFilter {
+  digest: Digest
+  sequenceNumber: U64
+}
+
+enum Order { ASC, DESC }
+
+input ObjectQuery {
+  # one of these
+  any: [ObjectQuery]
+  all: [ObjectQuery]
+  not: ObjectQuery
+
+  # (x)or any of these
+
+  # Cascading (type requires module requires package)
+  package: SuiAddress
+  module: String
+  type: String
+
+  owner: SuiAddress
+  objectId: SuiAddress
+  version: U64
+}
+
+input EventFilter {
+  # one of these
+  any: [EventFilter]
+  all: [EventFilter]
+  not: EventFilter
+
+  # (x)or any of these
+  sender: SuiAddress
+  transaction: Digest
+
+  # Cascading (module requires package)
+  emittingPackage: SuiAddress
+  emittingModule: String
+
+  # Cascading (type requires module requires package)
+  eventPackage: SuiAddress
+  eventModule: String
+  eventType: String
+
+  eventField: EventField
+
+  startTimeMs: U64
+  endTimeMs: U64
+}
+
+input EventField {
+  path: String!
+  value: JSON!
+}
+
+input TransactionBlockFilter {
+  # one of these
+  any: [TransactionBlockFilter]
+  all: [TransactionBlockFilter]
+  not: TransactionBlockFilter
+
+  # (x)or any of these
+
+  # Cascading (function requires module requires package)
+  package: SuiAddress
+  module: String
+  function: String
+
+  kind: TransactionBlockKindInput
+  checkpoint: U64
+  fromAddress: SuiAddress
+  toAddress: SuiAddress
+
+  inputObject: SuiAddress
+  changedObject: SuiAddress
+}
+
+type ConsistentRange {
+  first: Checkpoint
+  last: Checkpoint
+}
+
+interface Owner {
+  id: SuiAddress!
+
+  objects(query: ObjectQuery): ObjectQueryConnection
+
+  balance(type: TypeInst!): Balance
+  balances: BalanceConnection
+
+  coins(type: TypeInst): CoinConnection
+
+  stakes: StakeConnection
+
+  defaultNameServiceName: String
+  nameServiceNames: NameServiceNamesConnection
+}
+
+# Returned by Object.owner, where we can't disambiguate between
+# Address and Object.
+type AmbiguousOwner implements Owner {}
+
+type Address implements Owner {
+  transactionBlocks(
+    filter: AddressTransactionBlockFilter,
+  ): TransactionBlockConnection
+}
+
+enum AddressTransactionBlockFilter {
+  SIGN # Transactions this address has signed
+  SENT # Transactions that transferred objects from this address
+  RECV # Transactions that received objects into this address
+  PAID # Transactions that were paid for by this address
+}
+
+type Object implements Owner {
+  owner: AmbiguousOwner
+  version: U64
+  digest: Digest
+
+  previousVersion: U64
+  previousTransactionBlock: TransactionBlock
+  storageRebate: U64
+
+  display: [DisplayEntry]
+
+  asMoveObject: MoveObject
+  asMovePackage: MovePackage
+
+  # Transaction Blocks that sent objects to this object
+  receivedTransactionBlocks: TransactionBlockConnection
+
+  dynamicField(dynamicFieldName: DynamicFieldName!): DynamicField
+  dynamicFields: DynamicFieldConnection
+
+  bcs: Bytes
+}
+
+type DisplayEntry {
+  key: String
+  value: String
+}
+
+type Epoch {
+  epochId: Int!
```

Comment:
Epoch and generally anything that is counted in increments of epochs uses the `Int` type rather than `U64`, for pragmatic reasons -- we're not going to overflow 2^53 epochs any time soon.